include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -888,6 +888,11 @@ inline void format_bam::write_alignment_record([[maybe_unused]] stream_type &  s
         }
         else
         {
+            if (std::ranges::distance(qual) != core.l_seq)
+                throw format_error{detail::to_string("Expected quality of same length as sequence with size ",
+                                                     core.l_seq, ". Got quality with size ",
+                                                     std::ranges::distance(qual), " instead.")};
+
             assert(static_cast<int32_t>(std::ranges::distance(qual)) == core.l_seq);
```

Comment:
then remove the assert

Reply:
Well, I left it because it doesn't hurt. But I don't mind removing it.

Reply:
But it will never be evaluated because the exception is thrown before. So it is unreachable anyway